### DISCUSSION TITLE
How to configure multiple inline handlebar functions

### DISCUSSION THREAD

**Discussion by SWarnberg** [rfennell/AzurePipelines#1166] [General]
Hi,

I've struggling to configure multiple JavaScript functions as inline code in the GenerateReleaseNotes task in my release definition, but it seems it's not supported, or is it?
Having the following code (removed all line breaks too):

`customHandlebarsExtensionCode: 'module.exports = { getReleaseModule: function () {return process.env.ReleaseModule;}; getReleaseVersion: function () {return process.env.BUILD_VERSION;};};'`

I get either "Unexpected token ;" errors if I have bracket after function, or "Unexpected identifier" if I have not.

**Comment by rfennell** [comment]
Is it not easier to include them as a file?

Try the `customHandlebarsExtensionFile` parameter

**Comment by SWarnberg** [comment]
I can answer my own question. I found the problem. The example here https://github.com/rfennell/AzurePipelines/tree/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203#custom-extensions is not correct, at least not for inline scripts. It has to be a comma that separates the functions and not a semicolon.

**Reply by rfennell** [reply]
So it should be as follows?

```
const handlebars = require("handlebars");
module.exports = {
    count_workitems_by_type: function (array, typeName) {
        return array.filter(wi => wi.fields['System.WorkItemType'] === typeName).length;
    },
    replace_text: function (msg, match, replacement) {
        return msg.replace(match, replacement);
    },
};
```

**Reply by SWarnberg** [reply]
Exactly. Except, the last comma after the second function is not needed.

**Reply by rfennell** [reply]
I will update the document. Thanks for pointing it out

**Reply by rfennell** [reply]
Just published the updated docs
